forge-game/src/main/java/forge/game/GameAction.java

```diff
@@ -365,18 +365,31 @@ private Card changeZone(final Zone zoneFrom, Zone zoneTo, final Card c, Integer
                 copied.getOwner().removeInboundToken(copied);
 
                 if (repres == ReplacementResult.Prevented) {
-                    if (game.getStack().isResolving(c) && !zoneTo.is(ZoneType.Graveyard)) {
+                    c.clearEtbCounters();
+                    if (cause != null) {
+                        if (cause.hasParam("AnimateSubAbility")) {
+                            c.removeChangedCardKeywords(game.getTimestamp(), 0);
```

Comment:
i don't know if that is going to have the right timestamps

Reply:
You're right, some corner case might change it between
See what you think of this approach

Reply:
imo that feels more save for now